### PR TITLE
Copy dns requests and give them a fresh id when doing recursive queries.

### LINKED ISSUE #1113: Investigate if its safe to reuse request for recursive DNS
I'm not sure you can just send the same request we get from the client to the upstream server. This could lead to a conflict if two clients send a request with the same message ID. I would make a copy...

### PR DESCRIPTION
**Pull request by tomwilkie**
Fixes #1113

### CODE REVIEW

**Comment by tomwilkie** [comment]
@inercia would you mind reviewing please?

File: nameserver/dns.go [hunk 1]
```diff
@@ -227,7 +227,9 @@ func (d *DNSServer) handleRecursive(client *dns.Client, defaultMaxResponseSize i
 		}
 
 		for _, server := range d.upstream.Servers {
-			response, _, err := client.Exchange(req, fmt.Sprintf("%s:%s", server, d.upstream.Port))
+			reqCopy := *req
```
Comment (inercia) [inline nameserver/dns.go]: There is a `Copy()` method for messages...

Comment (tomwilkie) [inline nameserver/dns.go]: Ah good catch!  I was looking for it and I didn't find one.  Do you know where I could find package level docs for the dns stuff?

Comment (inercia) [inline nameserver/dns.go]: https://godoc.org/github.com/miekg/dns ?

**Comment by inercia** [comment]
LGTM. Will merge once the CI tests pass...

File: nameserver/dns.go (hunk 1 above)
Comment (tomwilkie) [inline nameserver/dns.go]: Sweet, thanks.

**Pull request merged**